README.md

```diff
@@ -1,35 +1,254 @@
+
+---
+
+<div align="center"><h2 style="font-weight:bold">What does it do?</h2></div>
+
+In a nutshell, this pulls versions out of XML data files for use in workflows.
+Just tell the action which repo, branch, and file contains the version, and it will search through the file for the `version-keys` and pull out the value of that key. This value is used as the value of the action's output,
+which has the name `version`, so you can use it in the rest of your workflow.
+
+The `version-keys` input is just a comma delimited list of XML keys to search for in the XML file.  
+Example:  
+If the value of the `version-keys` input was ***"Version,FileVersion"***, than it would search
+the XML for any XML elements that match the name ***"Version"*** or ***"FileVersion"***.  The first element that has a value will be the value returned.  So if the XML element ***"Version"*** had a value of ***1.2.3***, than it would simply return the value of the ***"Version"*** element and stop looking for values in any other XML elements.
+
+---
+
+<div align="center"><h3 style="font-weight:bold">Quick Example</h3></div>
+
+```yaml
+name: Get Version Example
+
+jobs:
+  Get_Version_Job:
+    runs-on: ubuntu-latest # Cannot use windows
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Get Version From C# Project File
+      id: get-version
+      uses: KinsonDigital/VersionMiner@v1.0.0-preview.1
+      with:
+      repo-owner: JohnDoe
+      repo-name: MyRepo
+      branch-name: master
+      file-type: xml
+      file-path: MyProject/MyProject.csproj
+      version-keys: Version
+
+    - name: Print Version From File
+      id: print-output
+      run: echo "${{ steps.get-version.outputs.version }}"
 ```
-{
-    "environment": "",
-    "repoOwner": "",
-	"repoName": "",
-	"relativeFilePath": "",
-	"userName": "",
-	"password": ""
-}
+
+So if the C# project file had the contents below, the workflow above would print the value ***1.2.3*** to the GitHub console.
+
+``` xml
+<Project Sdk="Microsoft.NET.Sdk">
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>10.0</LangVersion>
+    <Version>1.2.3</Version>
+    <FileVersion>0.1.0</FileVersion>
+</Project>
 ```
 
-## TypeScript Linting
+<div align="left">
+<a href="#examples">More Examples Below!! 👇🏼</a>
+</div>
+
+---
+
+<div align="center">
+
+## **Action Inputs**
+</div>
+
+| Input Name | Description | Required | Default Value |
+|---|:----|:---:|---|
+| `repo-owner` | The owner of the repository | yes | N/A |
+| `repo-name` | The name of the repository | yes | N/A |
+| `branch-name` | The name of the branch where the file lives. | yes | N/A |
+| `file-format` | The type of file that contains the version. Currently, the only supported value is `xml`. | yes | N/A |
+| `file-path` | The path to the file relative to the root of the repository. | yes | N/A |
+| `version-keys` | The key(s) that can hold the version in the file. | yes | N/A |
```

Comment:
So I have a question.   
Should I have it be 

```suggestion
| `version-keys` | The key(s) that possibly hold the version in the file. | yes | N/A |
```
or 
```suggestion
| `version-keys` | The key(s) that might hold the version in the file. | yes | N/A |
```
or just leave it as is?

The reason I ask this is because the key might not actually hold a value.  It might be acceptable for somebody to have it not hold a value and the action is built in a way to let the user choose if they want to force the value to exist or not.

Reply:
"Can" is the better word to use.